src/Compilers/Core/Portable/Text/SourceText.cs

```diff
@@ -280,14 +283,18 @@ private static string Decode(Stream stream, Encoding encoding, out Encoding actu
         {
             RoslynDebug.Assert(stream != null);
             RoslynDebug.Assert(encoding != null);
+            int length = 4096;
```

Comment:
>length [](start = 16, length = 6)

Consider creating a local called `bufferSize` instead since this does not represent the length of the buffer in the `!CanSeek` case. Then `int length;` and `bufferSize = Math.Min(4096, length);` can be moved inside the `if` block.